### ISSUE TITLE
function inline

### ISSUE THREAD

**Issue by Horcrux7** [i-net-software/JWebAssembly#10]
The current compiler of Wasm code to native machine code does not have optimizing like the JIT in Java. This means there is no code optimizing like the inline of functions. That JWebAssembly must already do this function inline at creating of the Wasm file.

Code samples that should JWebAssembly should optimize are welcome.

**Comment by Frontrider** [comment]
It *could* piggiback on this tool, used by a lot of the compilers.
https://github.com/WebAssembly/binaryen

There is a wasm to wasm optimizer in there.

**Comment by Frontrider** [comment]
#9 may be related.

**Comment by Horcrux7** [comment]
Yes, this tool can show some optimization. But I think this tool does not know any over the original program. The optimization will be limited.

**Comment by Frontrider** [comment]
Welp. I hope it still matters for something. It can also have an interesting side effect where you could use this tool as a java to javascript transpiler, by going through webassembly.
